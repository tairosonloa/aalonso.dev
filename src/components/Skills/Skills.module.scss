@import 'src/styles/utils.scss';
@import 'src/styles/palette.scss';

.container {
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  padding: grid_units(17) grid_units(4);
  background-color: $white;
  text-align: center;
}

.header {
  padding-bottom: grid_units(8);
}

.skillsContainer {
  display: grid;
  grid-gap: grid_units(15);
  grid-template-columns: repeat(auto-fit, minmax(grid_units(50), 1fr));
  align-items: start;
  width: 100%;
  max-width: grid_units(200);
  padding-right: grid_units(4);
  padding-left: grid_units(4);
  text-align: center;
}

.headIcon {
  color: $burgundy;
  font-size: 40px;
}

.skillHeaderText {
  margin-top: grid_units(2);
  margin-bottom: grid_units(4);
}

.skillsLogos {
  display: flex;
  flex-direction: row;
  flex-wrap: wrap;
  justify-content: center;
  width: 100%;

  & > div {
    margin: grid_units(2);
  }
}

.skillIcon {
  margin-bottom: grid_units(1);
  color: $burgundy;
  font-size: 35px;
}

.iconWithDescription {
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
}
