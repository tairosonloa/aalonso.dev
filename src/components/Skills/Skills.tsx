import React, { FC } from 'react'
import { DevIcon, DevIconName } from '../DevIcon/DevIcon'
import { Icon, IconName } from '../Icon/Icon'
import { Text, TextType } from '../Text/Text'
import styles from './Skills.module.scss'

const techSkills = [
  { icon: DevIconName.REACT, name: 'React' },
  { icon: DevIconName.NODE_JS, name: 'Node.js' },
  { icon: DevIconName.TYPESCRIPT, name: 'TypeScript' },
  { icon: DevIconName.HTML, name: 'HTML 5' },
  { icon: DevIconName.CSS, name: 'CSS 3' },
  { icon: DevIconName.PYTHON, name: 'Python 3' },
  { icon: DevIconName.POSTGRESQL, name: 'PostgreSQL' },
  { icon: DevIconName.MONGODB, name: 'MongoDB' },
  { icon: DevIconName.LINUX, name: 'Linux' },
  { icon: DevIconName.DOCKER, name: 'Docker' },
  { icon: DevIconName.AWS, name: 'Amazon AWS' },
]

const softSkills = [
  { icon: IconName.TEAM_WORK, name: 'Teamwork' },
  { icon: IconName.COMMUNICATION, name: 'Communication' },
  { icon: IconName.EMPATHY, name: 'Empathy' },
  { icon: IconName.CURIOSITY, name: 'Curiosity' },
  { icon: IconName.TIME_MANAGEMENT, name: 'Time management' },
  { icon: IconName.PROBLEM_SOLVING, name: 'Problem solving' },
  { icon: IconName.ACCOUNTABILITY, name: 'Accountability' },
  { icon: IconName.OPEN_MINDEDNESS, name: 'Open-mindedness' },
  { icon: IconName.CREATIVITY, name: 'Creativity' },
  { icon: IconName.ADAPTABILITY, name: 'Adaptability' },
  { icon: IconName.PROACTIVITY, name: 'Proactivity' },
]

export const Skills: FC = () => {
  return (
    <div className={styles.container}>
      <div className={styles.header}>
        <Text textType={TextType.TITLE}>What I offer</Text>
        <Text textType={TextType.TITLE_SMALL}>Why you might want to work with me.</Text>
      </div>
      <div className={styles.skillsContainer}>
        <div data-testid="tech-skills">
          <Icon name={IconName.TECH_SKILLS} className={styles.headIcon} />
          <Text textType={TextType.PARAGRAPH_LEAD} className={styles.skillHeaderText}>
            Tech Skills
          </Text>
          <div className={styles.skillsLogos}>
            {techSkills.map((skill) => (
              <div className={styles.iconWithDescription} key={skill.name}>
                <DevIcon name={skill.icon} className={styles.skillIcon} />
                <span>{skill.name}</span>
              </div>
            ))}
          </div>
        </div>
        <div data-testid="soft-skills">
          <Icon name={IconName.SOFT_SKILLS} className={styles.headIcon} />
          <Text textType={TextType.PARAGRAPH_LEAD} className={styles.skillHeaderText}>
            Soft Skills
          </Text>
          <div className={styles.skillsLogos}>
            {softSkills.map((skill) => (
              <div className={styles.iconWithDescription} key={skill.name}>
                <Icon name={skill.icon} className={styles.skillIcon} />
                <span>{skill.name}</span>
              </div>
            ))}
          </div>
        </div>
      </div>
    </div>
  )
}
