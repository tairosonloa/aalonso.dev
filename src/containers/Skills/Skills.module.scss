@import 'src/styles/utils.scss';
@import 'src/styles/palette.scss';

.skillsContainer {
  display: grid;
  grid-gap: grid_units(15);
  grid-template-columns: repeat(auto-fit, minmax(grid_units(76), 1fr));
  align-items: start;
  width: 100%;
  max-width: grid_units(200);
  padding-right: grid_units(4);
  padding-left: grid_units(4);
  text-align: center;

  @include screenIn(mobile) {
    grid-template-columns: repeat(auto-fit, minmax(grid_units(56), 1fr));
  }
}

.headIcon {
  color: $burgundy;
  font-size: 40px;
}

.skillHeaderText {
  margin-top: grid_units(2);
  margin-bottom: grid_units(4);
}

.skillsLogos {
  display: flex;
  flex-direction: row;
  flex-wrap: wrap;
  align-items: center;
  justify-content: center;
  width: 100%;
  max-width: grid_units(100);
  margin: auto;

  & > div {
    margin: grid_units(2);
  }
}

.skillIcon {
  margin-bottom: grid_units(1);
  color: $burgundy;
  font-size: 35px;
}

.iconWithDescription {
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
}
