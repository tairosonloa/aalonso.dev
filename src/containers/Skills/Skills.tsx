import React, { FC } from 'react'
import { SectionContainer } from '../../components/DataDisplays/SectionContainer/SectionContainer'
import { SectionHeader } from '../../components/DataDisplays/SectionHeader/SectionHeader'
import { Text, TextType } from '../../components/DataDisplays/Text/Text'
import { DevIcon, DevIconName } from '../../components/Icons/DevIcon/DevIcon'
import { IconName, RemixIcon } from '../../components/Icons/RemixIcon/RemixIcon'
import styles from './Skills.module.scss'

const techSkills = [
  { icon: DevIconName.REACT, name: 'React' },
  { icon: DevIconName.NODE_JS, name: 'Node.js' },
  { icon: DevIconName.TYPESCRIPT, name: 'TypeScript' },
  { icon: DevIconName.HTML, name: 'HTML 5' },
  { icon: DevIconName.CSS, name: 'CSS 3' },
  { icon: DevIconName.PYTHON, name: 'Python 3' },
  { icon: DevIconName.POSTGRESQL, name: 'PostgreSQL' },
  { icon: DevIconName.MONGODB, name: 'MongoDB' },
  { icon: DevIconName.LINUX, name: 'Linux' },
  { icon: DevIconName.DOCKER, name: 'Docker' },
  { icon: DevIconName.AWS, name: 'Amazon AWS' },
]

const softSkills = [
  { icon: IconName.TEAM_WORK, name: 'Teamwork' },
  { icon: IconName.COMMUNICATION, name: 'Communication' },
  { icon: IconName.EMPATHY, name: 'Empathy' },
  { icon: IconName.CURIOSITY, name: 'Curiosity' },
  { icon: IconName.TIME_MANAGEMENT, name: 'Time management' },
  { icon: IconName.PROBLEM_SOLVING, name: 'Problem solving' },
  { icon: IconName.ACCOUNTABILITY, name: 'Accountability' },
  { icon: IconName.OPEN_MINDEDNESS, name: 'Open-mindedness' },
  { icon: IconName.CREATIVITY, name: 'Creativity' },
  { icon: IconName.ADAPTABILITY, name: 'Adaptability' },
  { icon: IconName.PROACTIVITY, name: 'Proactivity' },
]

export const Skills: FC = () => {
  return (
    <SectionContainer>
      <SectionHeader title="What I offer" subtitle="Why you might want to work with me." />
      <div className={styles.skillsContainer}>
        <div data-testid="tech-skills">
          <RemixIcon name={IconName.TECH_SKILLS} className={styles.headIcon} />
          <Text textType={TextType.PARAGRAPH_LEAD} className={styles.skillHeaderText}>
            Tech Skills
          </Text>
          <div className={styles.skillsLogos}>
            {techSkills.map((skill) => (
              <div className={styles.iconWithDescription} key={skill.name}>
                <DevIcon name={skill.icon} className={styles.skillIcon} />
                <span>{skill.name}</span>
              </div>
            ))}
          </div>
        </div>
        <div data-testid="soft-skills">
          <RemixIcon name={IconName.SOFT_SKILLS} className={styles.headIcon} />
          <Text textType={TextType.PARAGRAPH_LEAD} className={styles.skillHeaderText}>
            Soft Skills
          </Text>
          <div className={styles.skillsLogos}>
            {softSkills.map((skill) => (
              <div className={styles.iconWithDescription} key={skill.name}>
                <RemixIcon name={skill.icon} className={styles.skillIcon} />
                <span>{skill.name}</span>
              </div>
            ))}
          </div>
        </div>
      </div>
    </SectionContainer>
  )
}
