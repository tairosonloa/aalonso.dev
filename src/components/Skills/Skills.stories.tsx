import React from 'react'
import { dimensionsDecorator } from '../../../.storybook/utils/Dimensions'
import { Skills } from './Skills'

export default {
  title: 'Common|Skills',
  component: Skills,
  decorators: [dimensionsDecorator()],
}

export const Default = (): JSX.Element => <Skills />
