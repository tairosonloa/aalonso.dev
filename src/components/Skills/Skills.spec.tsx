import { render } from '@testing-library/react'
import * as React from 'react'
import { Skills } from './Skills'

describe('<Skills/>', () => {
  it('should display the default message', () => {
    const title = 'What I offer'
    const subtitle = 'Why you might want to work with me.'

    const renderResult = render(<Skills />)

    expect(renderResult.getByText(title)).toBeTruthy()
    expect(renderResult.getByText(subtitle)).toBeTruthy()
  })

  describe('tech skills', () => {
    it('should display the section title', () => {
      const title = 'Tech Skills'

      const renderResult = render(<Skills />)

      expect(renderResult.getByText(title)).toBeTruthy()
    })

    it('should display math the snpashot', () => {
      const testId = 'tech-skills'

      const renderResult = render(<Skills />)

      expect(renderResult.getByTestId(testId)).toMatchSnapshot()
    })
  })

  describe('soft skills', () => {
    it('should display the section title', () => {
      const title = 'Soft Skills'

      const renderResult = render(<Skills />)

      expect(renderResult.getByText(title)).toBeTruthy()
    })

    it('should display math the snpashot', () => {
      const testId = 'soft-skills'

      const renderResult = render(<Skills />)

      expect(renderResult.getByTestId(testId)).toMatchSnapshot()
    })
  })
})
