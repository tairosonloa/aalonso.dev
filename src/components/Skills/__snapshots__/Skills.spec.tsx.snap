// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Skills/> soft skills should display math the snpashot 1`] = `
<div
  data-testid="soft-skills"
>
  <i
    class="ri-hand-heart-fill icon headIcon"
  />
  <h4
    class="common paragraphLead skillHeaderText"
  >
    Soft Skills
  </h4>
  <div
    class="skillsLogos"
  >
    <div
      class="iconWithDescription"
    >
      <i
        class="ri-team-fill icon skillIcon"
      />
      <span>
        Teamwork
      </span>
    </div>
    <div
      class="iconWithDescription"
    >
      <i
        class="ri-chat-smile-2-fill icon skillIcon"
      />
      <span>
        Communication
      </span>
    </div>
    <div
      class="iconWithDescription"
    >
      <i
        class="ri-heart-fill icon skillIcon"
      />
      <span>
        Empathy
      </span>
    </div>
    <div
      class="iconWithDescription"
    >
      <i
        class="ri-questionnaire-fill icon skillIcon"
      />
      <span>
        Curiosity
      </span>
    </div>
    <div
      class="iconWithDescription"
    >
      <i
        class="ri-timer-line icon skillIcon"
      />
      <span>
        Time management
      </span>
    </div>
    <div
      class="iconWithDescription"
    >
      <i
        class="ri-pencil-ruler-2-line icon skillIcon"
      />
      <span>
        Problem solving
      </span>
    </div>
    <div
      class="iconWithDescription"
    >
      <i
        class="ri-scales-3-line icon skillIcon"
      />
      <span>
        Accountability
      </span>
    </div>
    <div
      class="iconWithDescription"
    >
      <i
        class="ri-earth-fill icon skillIcon"
      />
      <span>
        Open-mindedness
      </span>
    </div>
    <div
      class="iconWithDescription"
    >
      <i
        class="ri-lightbulb-line icon skillIcon"
      />
      <span>
        Creativity
      </span>
    </div>
    <div
      class="iconWithDescription"
    >
      <i
        class="ri-seedling-fill icon skillIcon"
      />
      <span>
        Adaptability
      </span>
    </div>
    <div
      class="iconWithDescription"
    >
      <i
        class="ri-run-fill icon skillIcon"
      />
      <span>
        Proactivity
      </span>
    </div>
  </div>
</div>
`;

exports[`<Skills/> tech skills should display math the snpashot 1`] = `
<div
  data-testid="tech-skills"
>
  <i
    class="ri-code-box-fill icon headIcon"
  />
  <h4
    class="common paragraphLead skillHeaderText"
  >
    Tech Skills
  </h4>
  <div
    class="skillsLogos"
  >
    <div
      class="iconWithDescription"
    >
      <i
        class="devicon-react-original icon skillIcon"
      />
      <span>
        React
      </span>
    </div>
    <div
      class="iconWithDescription"
    >
      <i
        class="devicon-nodejs-plain icon skillIcon"
      />
      <span>
        Node.js
      </span>
    </div>
    <div
      class="iconWithDescription"
    >
      <i
        class="devicon-typescript-plain icon skillIcon"
      />
      <span>
        TypeScript
      </span>
    </div>
    <div
      class="iconWithDescription"
    >
      <i
        class="devicon-html5-plain icon skillIcon"
      />
      <span>
        HTML 5
      </span>
    </div>
    <div
      class="iconWithDescription"
    >
      <i
        class="devicon-css3-plain icon skillIcon"
      />
      <span>
        CSS 3
      </span>
    </div>
    <div
      class="iconWithDescription"
    >
      <i
        class="devicon-python-plain icon skillIcon"
      />
      <span>
        Python 3
      </span>
    </div>
    <div
      class="iconWithDescription"
    >
      <i
        class="devicon-postgresql-plain icon skillIcon"
      />
      <span>
        PostgreSQL
      </span>
    </div>
    <div
      class="iconWithDescription"
    >
      <i
        class="devicon-mongodb-plain icon skillIcon"
      />
      <span>
        MongoDB
      </span>
    </div>
    <div
      class="iconWithDescription"
    >
      <i
        class="devicon-linux-plain icon skillIcon"
      />
      <span>
        Linux
      </span>
    </div>
    <div
      class="iconWithDescription"
    >
      <i
        class="devicon-docker-plain icon skillIcon"
      />
      <span>
        Docker
      </span>
    </div>
    <div
      class="iconWithDescription"
    >
      <i
        class="devicon-amazonwebservices-plain icon skillIcon"
      />
      <span>
        Amazon AWS
      </span>
    </div>
  </div>
</div>
`;
